spec.emu

```diff
@@ -86,15 +133,16 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
         <h1>The _NativeError_ Constructors</h1>
 
         <emu-clause id="sec-nativeerror">
-          <h1>_NativeError_ ( _message_<ins> [ , _cause_ ]</ins> )</h1>
-          <p>When a _NativeError_ function is called with argument _message_<ins> and _cause_</ins>, the following steps are taken:</p>
+          <h1>_NativeError_ ( _message_<ins> [ , _options_ ]</ins> )</h1>
+          <p>When a _NativeError_ function is called with argument _message_<ins> and _options_</ins>, the following steps are taken:</p>
           <emu-alg>
             1. If NewTarget is *undefined*, let _newTarget_ be the active function object; else let _newTarget_ be NewTarget.
             1. [id="step-nativerror-ordinarycreatefromconstructor"] Let _O_ be ? OrdinaryCreateFromConstructor(_newTarget_, <code>"%<var>NativeError</var>.prototype%"</code>, &laquo; [[ErrorData]] &raquo;).
             1. If _message_ is not *undefined*, then
               1. Let _msg_ be ? ToString(_message_).
               1. Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
               1. Perform ! DefinePropertyOrThrow(_O_, *"message"*, _msgDesc_).
+            1. <ins>Let _cause_ be ? GetOption(_options_, `"cause"`)</ins>
             1. <ins>If _cause_ is not *undefined*, then</ins>
```

Comment:
It's possible to throw `undefined`. This check means that if you have

```js
try {
  // something which throws `undefined`
} catch (e) {
  throw new Error('whoops', { cause: e });
}
```
will result in an error which does not have a `cause` property. That seems wrong.

Instead, we should create the property if and only if `_options_` is present, and is an object, and `? HasProperty(_options_, "cause")` is `*true*`.